### DISCUSSION TITLE
Hoek-Brown model and Equivalent plastic strain

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#17892] [Q&A Modules: Solid mechanics] [answered]
Hi guys,

just wonder does MOOSE have the Hoek-Brown plastic model, I couldn't find it in the document at all (I only see DP and MC models), maybe it's hidden somewhere?

Do we have the 'equivalent plastic strain' or the 'Damage' output for the tensor mechanics module? I guess it's this (below), someone please help me to confirm it.

```
  [./mc_int_auxk]
    type = MaterialStdVectorAux
    index = 0
    property = plastic_internal_parameter
    variable = mc_int
  [../]
```

Thank you guys, I hope you have a good day
Traiwit

**Comment by WilkAndy** [comment, marked as answer]
No Hoek-Brown that i'm aware of.  (I wonder whether a private App has it?)

You are right about the MaterialStdVectorAux.  You just have to be careful with the `index`, because that depends on your plastic model.  If you are unsure, it's easiest to consult the test files that use the same plastic model that you're using, because they almost always have record plastic strain into AuxVars.  If still unsure, just ask.

a

**Reply by Traiwit** [reply]
Hi @WilkAndy, my team has implemented Hoek-Brown in Abaqus, not sure how difficult it is to transfer into MOOSE.

Regarding 'index', I checked several input files, they are all over the place, could you please explain it to me.

If it helps, most of my simulations consider Z-axis to be verticle direction. Should the index be '2' OR I am on the wrong track entirely?

Thank you
Traiwit

**Reply by GiudGiud** [reply]
Any news on this @Traiwit @WilkAndy

**Reply by WilkAndy** [reply]
Thanks for prompting, @GiudGiud .  

The "index" is nothing to do with direction, @Traiwit .   It is dependent on the plastic model.  For instance, some implementations of Mohr-Coulomb will have 6 yield functions, with each of those functions having a plastic strain associated with them.  Therefore, when using this type of MohrCoulomb you'll have to choose index=0 or index=1, ... or index=5.   On the other hand, another implementation of Mohr-Coulomb will have just 1 yield function (perhaps it has smoothed the 6 mentioned above into one overall surface, or perhaps it does some rearrangement of the principal stresses, or something else) and will have just 1 plastic strain.  Therefore, your only choice is index=0.

So, in conclusion, you have to know the details of the particular implementation of your particular plastic model in order to know what "index" you might be interested in recording.

I suggest you just ask this Discussion list when you're not sure.   Or you can "risk it" and look at test files that use the exact plastic model you're using.

**Reply by Traiwit** [reply]
Thanks @WilkAndy 

Now i'm more confused with the '6' yield functions, can we control it in the input file?

I think the one I'm using doesn't have the option `(TensorMechanicsPlasticMohrCoulombMulti)`, so index = 0 should be fine

![Capture](https://user-images.githubusercontent.com/79562439/124848367-0b4a0200-dfe0-11eb-9251-8cf124e40bef.PNG)

Thanks again guys

**Reply by WilkAndy** [reply]
You are correct about `index=0` @Traiwit .   `TensorMechanicsPlasticMohrCoulombMulti` does have 6 planar yield functions, so could have 6 plastic internal parameters, but MCMulti actually sums them, so there is only 1.  Hence `index=0` is the only valid choice.

(This is not true for other implementations of Mohr-Coulomb.  For instance `CappedMohrCoulombStressUpdate` has two internal parameters, `index=0` corresponding to shear failure and `index=1` corresponding to tensile failure.  BTW, it has 12 yield functions, the first 3 being tensile, the next 3 being compression, the next 6 being Mohr-Coulomb.)

For future reference, when you're using the `ComputeMultiPlasticityStress` with, say
```
plastic_models = 'model0 model1 model2'
```
then
- `index=0` will correspond to the plastic internal parameter of model0
- `index=1` will correspond to the plastic internal parameter of model1
- `index=2` will correspond to the plastic internal parameter of model2

**Reply by xxsszzaa** [reply]
> Hi @WilkAndy, my team has implemented Hoek-Brown in Abaqus, not sure how difficult it is to transfer into MOOSE.
> 
> Regarding 'index', I checked several input files, they are all over the place, could you please explain it to me.
> 
> If it helps, most of my simulations consider Z-axis to be verticle direction. Should the index be '2' OR I am on the wrong track entirely?
> 
> Thank you Traiwit

Hi @Traiwit ，Recently, I want to use the material subroutines of Hoke-Brown and Mogi Coulomb for comparative tests. I wonder if you can send me the source code of your 's UMAT, 850082649@qq.com 

Thank you very much

Sandwish
